include/navigation/ridf_processors.h

```diff
@@ -42,7 +42,7 @@ struct BestMatchingSnapshot
         const float difference = minDifferences[bestSnapshot] / 255.0f;
 
         // Bundle result as tuple
-        return std::make_tuple(heading, bestSnapshot, difference);
+        return std::make_tuple(heading, bestSnapshot + startSnapshot, difference);
```

Comment:
Wait... Doesn't this mean that the index returned by ``getHeading()`` will be absolute rather than relative to the start of the window?

Reply:
It is indeed - it makes the general case easier as the user doesn't have to then add the window start to the best match index to update the window (I guess, that case could actually take relative headings quite happily) or for logging. It no longer matches up with stuff like ``m_RotatedDifferences`` though which leaves me a little torn on whether this is a good idea or not. Your thoughts would be appreciated!

Reply:
Any thought @alexdewar ?

Reply:
I have a feeling that the most elegant way to deal with this is to switch from returning a ``std::tuple`` to a ``class``, which could have getters that abstract away these sorts of details. I'm happy to have a look at this when I add tests and merge it in with the refactoring I've been doing...

For now I think it's ok. It doesn't break any existing code or anything.

Reply:
that almost always seems the best way of improving tuple-based code doesn't it :disappointed: 

Ok, cool, I'll test this on the robot this afternoon and merge if everything works

Reply:
Wicked. This is a nice enhancement.